### ISSUE TITLE
Validate of smart-flank-gcs-path to not override different results type

### ISSUE THREAD

**Issue by adamfilipow92** [Flank/flank#1057]
**Author the user story for this feature**

As a flank user, I want to be informed when flank can override JUnitResult configured in ```smart-flank-gcs-path``` with a different type in the current run so I can easily avoid situations like that 

**Is your feature request related to a problem? Please describe.**

If the test is flaky, flank removing stack traces from JUnitResult, FullJunitResult keeps stack traces.

**Describe the solution you'd like**

Fast fail when Flank can potentially override JUnitResult with different report type.

**Comment by pawelpasterz** [comment]
Just for my understanding. Now flank will upload to a bucket only if types match? (by types I understand full and normal results)
example:
I run my test suite with flank and received `JUnitResult`. But I am not glad about the info I received and want to have `FullJUnitResult`. So now, I need to manually remove a file from the bucket and after that, I am able to run flank with the new config?

**Comment by pawelpasterz** [comment]
Sorry, it's just too many sources of information for this issue 😄 
Based on https://github.com/Flank/flank/issues/936 and implementation in https://github.com/Flank/flank/pull/1056 I am trying to understand the actual case.

> For now, we can run smart flank without set --full-junit-result and --legacy-junit-result but if we use FullJUnitResult in gcs path and current run don't have set flag --full-junit-result flank override full junit result by junit result it's can be little confusing

If I get it correctly it means that currently flank will try to upload `FullJUnitResult` to thew bucket but actually this file won't be produced because `--full-junit-result` was not used and `JUnitResult` is created instead. So flank will upload `JUnitResult` but rename it to `FullJUnitResult` Am I correct?

> If user set different XML name in smart-flank-gcs-path flank shouldn't validate type.

Basically means - ah, yeah, you changed the name so probably you know what you are doing (no validation)

Thanks for understanding and patience :)

**Comment by adamfilipow92** [comment]
So 😄 
1. If we set ```smart-flank-gcs-path``` to ```.../FullJUnitResult.xml``` and we have turned on flag ```--full-junit-result``` file from ```smart-flank-gcs-path``` path will be overridden by JUnitResult.xml. We want to avoid that case. 
There is a small exception mentioned by you on my pr, about turn off validation when ```--smart-flank-disable-upload``` set. Thanks! 

2. About different XML names - right, if the name is changed by the user we don't validate the type. The user probably knows what he doing.
